graphix/clifford.py

```diff
@@ -34,58 +35,84 @@ class Domains:
     t_domain: set[int]
 
 
-class Clifford:
+class Clifford(Enum):
     """Clifford gate."""
 
-    def __init__(self, index: int):
-        self.__index = index
-
-    @property
-    def index(self) -> int:
-        """Return the index of the Clifford gate (inverse of clifford.get)."""
-        return self.__index
+    # MEMO: Cannot use ClassVar here
+    I: Clifford
+    X: Clifford
+    Y: Clifford
+    Z: Clifford
+    S: Clifford
+    SDG: Clifford
+    H: Clifford
+
+    _0 = 0
+    _1 = 1
+    _2 = 2
+    _3 = 3
+    _4 = 4
+    _5 = 5
+    _6 = 6
+    _7 = 7
+    _8 = 8
+    _9 = 9
+    _10 = 10
+    _11 = 11
+    _12 = 12
+    _13 = 13
+    _14 = 14
+    _15 = 15
+    _16 = 16
+    _17 = 17
+    _18 = 18
+    _19 = 19
+    _20 = 20
+    _21 = 21
+    _22 = 22
+    _23 = 23
 
     @property
-    def matrix(self) -> np.ndarray:
+    def matrix(self) -> npt.NDArray[np.complex128]:
         """Return the matrix of the Clifford gate."""
-        return CLIFFORD[self.__index]
+        return CLIFFORD[self.value]
 
     def __repr__(self) -> str:
         """Return the Clifford expression on the form of HSZ decomposition."""
         return " @ ".join([f"graphix.clifford.{gate}" for gate in self.hsz])
```

Comment:
The namespace has changed.

```suggestion
        return " @ ".join([f"Clifford.{gate}" for gate in self.hsz])
```

Reply:
Fixed in 4ff8e6830c82e75aaaa27e1908cf95b09acc7de4 .